### PR TITLE
Reduce usages of fwd/bwd flags in DefaultEdgeFilter constructor

### PR DESCRIPTION
**Pull request by ammagamma**
#1380 

Replacing the constructor calls in other branches can be done quite easily using this regex search&replace in IntelliJ: 
replace

### CODE REVIEW

**Comment by ammagamma** [comment]
Should we replace the default constructor calls with `DefaultEdgeFilter.allEdges` as well ? Maybe there should be a better name for it to distinguish it from `EdgeFilter.ALL_EDGES` ?

File: core/src/main/java/com/graphhopper/routing/util/DefaultEdgeFilter.java [hunk 1]
```diff
@@ -40,6 +40,18 @@ public DefaultEdgeFilter(FlagEncoder encoder, boolean bwd, boolean fwd) {
         this.fwd = fwd;
     }
```
Comment (karussell) [inline core/src/main/java/com/graphhopper/routing/util/DefaultEdgeFilter.java]: Should we remove this or even all constructors? IMO the order bwd+fwd is unintuitive

**Comment by karussell** [comment]
> Should we replace the default constructor calls with DefaultEdgeFilter.allEdges as well ? 

IMO yes, so we have no mixed constructor+factory usage. At least the bwd+fwd constructor

> Maybe there should be a better name for it to distinguish it from EdgeFilter.ALL_EDGES ?

IMO this is fine - or maybe add a javadoc comment what it does?

File: core/src/main/java/com/graphhopper/routing/util/DefaultEdgeFilter.java (hunk 1 above)
Comment (ammagamma) [reply core/src/main/java/com/graphhopper/routing/util/DefaultEdgeFilter.java]: I removed the default constructor and made the other one protected. We could make it private, too and instead add a getter for the flag encoder (this is currently needed for the PrepEdgeFilter in PrepareRoutingSubnetworks).

Comment (ammagamma) [reply core/src/main/java/com/graphhopper/routing/util/DefaultEdgeFilter.java]: Changed the oder bwd+fwd too

**Pull request merged**